libnmstate/state.py

```diff
@@ -159,6 +223,35 @@ def merge_interfaces(self, other_state):
             dict_update(other_state.interfaces[name], self.interfaces[name])
             self._ifaces_state[name] = other_state.interfaces[name]
 
+    def merge_route_config(self, other):
+        """
+        Merge routes from other to self.
+        If interface does exists in self, create an empty one.
+        """
+        origin_other = copy.deepcopy(other)
+        # Copy other to self if certain interface does not defined.
+        for iface_name, routes in six.viewitems(other.iface_routes):
+            if iface_name not in self.iface_routes:
+                self.routes.extend(routes)
```

Comment:
With interfaces, we merge the interface entry but not complement interfaces that have not been mentioned. It does not help us to do so.

So could you please remind/clarify why this is needed here? It looks like it complements routes which have not been mentioned by the user, so the question is what for.

Reply:
When certain route mentioned `eth1` but `eth1` is not in the `State.interfaces()`, that route will not
be stored in `eth1` NM profile.
That's why we need to at least mention it.